data/core/command.lua

```diff
@@ -1,25 +1,46 @@
 local core = require "core"
 local command = {}
 
+---A predicate function accepts arguments from `command.perform()` and evaluates to a boolean. </br>
+---If the function returns true, then the function associated with the command is executed.
+---
+---The predicate function can also return other values after the boolean, which will
+---be passed into the function associated with the command.
+---@alias core.command.predicate_function fun(...: any): boolean, ...
+
+---A predicate is a string, Object or a function that is used to determine whether a command should be
+---executed. It could be used to match the current view among many other things.
+---
+---If the predicate is a string, it is resolved into an `Object` via `require()`
+---and checked against the active view with `Object:extends()`. </br>
+---For example, `"core.docview"` will match any view that inherits from `DocView`. </br>
+---A `!` can be appended to the predicate to strictly match the current view via `Object:is()`,
+---instead of matching any view that inherits the predicate.
+---
+---If the predicate is a table, it it checked against the active view with `Object:extends()`.
+---Strict matching via `Object:is()` is not available.
+---
+---If the predicate is a function, it must be a predicate function - a function that takes in
+---any arguments that may be passed via `command.perform()`, and returns a boolean and any
+---other values for the function associated to the command.
+---@alias core.command.predicate string|core.object|core.command.predicate_function
+
+---The predicate and its associated function.
+---@class core.command.command
+---@field predicate core.command.predicate_function
+---@field perform fun(...: any)
+
+---@type { [string]: core.command.command }
 command.map = {}
 
 local always_true = function() return true end
```

Comment:
Should we set the type of this to `core.command.predicate_function`, just to specify its usage?

Reply:
I'm not sure if `perform` is of type `core.command.predicate_function`.

Reply:
I mean the `always_true` function. That one should be a `core.command.predicate_function` right?